dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04345ChangeMultiTreePrimaryKey.java

```diff
@@ -0,0 +1,55 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.startup.StartupTask;
+import com.dotmarketing.util.MaintenanceUtil;
+import com.dotmarketing.util.UUIDGenerator;
+import com.dotmarketing.util.UtilMethods;
+
+import java.util.*;
+
+public class Task04345ChangeMultiTreePrimaryKey extends AbstractJDBCStartupTask  {
```

Comment:
on mssql the ddl changes must be under non transactional connection, all UT are under transactional connection keep that in mind

Reply:
All the upgrade task run with the auto commit set to true

https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotmarketing/startup/AbstractJDBCStartupTask.java#L740